### PR TITLE
Improvements in Nebula LLM

### PR DESCRIPTION
**Pull request by toshish**
- Description: Added improvements in Nebula LLM to perform auto-retry; more generation parameters supported. Conversation is no longer required to be passed in the LLM object. Examples are updated.
  - Issue: N/A
  - Dependencies: N/A
  - Tag maintainer: @baskaryan 
  - Twitter handle: symbldotai

### CODE REVIEW

File: libs/langchain/langchain/llms/symblai_nebula.py [hunk 1]
```diff
@@ -68,33 +82,39 @@ def validate_environment(cls, values: Dict) -> Dict:
             DEFAULT_NEBULA_SERVICE_PATH,
         )
         nebula_api_key = get_from_dict_or_env(
-            values, "nebula_api_key", "NEBULA_SERVICE_API_KEY", ""
+            values, "nebula_api_key", "NEBULA_API_KEY", None
```
Comment (baskaryan) [inline libs/langchain/langchain/llms/symblai_nebula.py:85]: believe this will still error if given a default value of None

Comment (baskaryan) [reply libs/langchain/langchain/llms/symblai_nebula.py:85]: so you'd never hit the error message in L88

Comment (toshish) [reply libs/langchain/langchain/llms/symblai_nebula.py:85]: @baskaryan you're right. Will remove the line 88 check. I tested it, the error gets reported if the value is set to None.

Comment (toshish) [reply libs/langchain/langchain/llms/symblai_nebula.py:85]: @baskaryan its updated.

**Comment by baskaryan** [comment]
lgtm, thanks @toshish!

**Pull request merged**